### PR TITLE
Migrate RTN, HQQ and AWQ to Torch new 3.x API

### PR DESCRIPTION
**Pull request by yuwenzho**
## Type of Change

feature
API changed or not: yes

## Description

Migrate RTN, HQQ and AWQ to Torch new 3x API

RTN
```python
from neural_compressor.torch.quantization import get_default_rtn_config, prepare, convert
quant_config = get_default_rtn_config()
model= prepare(model, quant_config)
q_model = convert(model)
```

HQQ
```python
from neural_compressor.torch.quantization import get_default_hqq_config, prepare, convert
quant_config = get_default_hqq_config()
model= prepare(model, quant_config)
q_model = convert(model)
```

AWQ
```python
from neural_compressor.torch.quantization import get_default_awq_config, prepare, convert
quant_config = get_default_awq_config()
# prepare
model= prepare(model, quant_config, example_inputs=example_inputs)
# calibrate
calib_func(model)
# convert
q_model = convert(model)
```

## How has this PR been tested?

CI

## Dependency Change?

no

### CODE REVIEW

File: test/3x/torch/quantization/weight_only/hqq/test_hqq_cuda.py [hunk 1]
```diff
@@ -110,6 +110,62 @@ def test_hqq_module_cuda(
...
+            (8, 64, False, False, 128),
+            (8, 64, True, True, 128),
+            (8, 64, False, True, 128),
+            (4, 64, True, False, 64),
+            (4, 64, False, False, 64),
+            (4, 64, True, True, 64),
+            (4, 64, False, True, 64),
+        ],
+    )
+    def test_hqq_module_cuda(
```
Comment (yiliu30) [inline test/3x/torch/quantization/weight_only/hqq/test_hqq_cuda.py:152]: Please run these on a machine with GPU manually to make sure it works.

**Review by yiliu30 — APPROVED** [review]
Others are LTGM.

**Review by Kaihui-intel — APPROVED** [review]

**Review by xin3he — APPROVED** [review]

File: test/3x/torch/quantization/weight_only/hqq/test_hqq_cuda.py (hunk 1 above)
Comment (yuwenzho) [reply test/3x/torch/quantization/weight_only/hqq/test_hqq_cuda.py:152]: Require #1767

Comment (yuwenzho) [reply test/3x/torch/quantization/weight_only/hqq/test_hqq_cuda.py:152]: Tested on local GPU machine and it passed.

**Pull request merged**